t/mojo/dom.t

```diff
@@ -2643,10 +2643,33 @@ subtest 'Comments' => sub {
 <!-- HTML4 -- >
 <!-- bad idea -- HTML4 -- >
 EOF
-  is $dom->tree->[1][1], ' HTML5 ',             'right comment';
-  is $dom->tree->[3][1], ' bad idea -- HTML5 ', 'right comment';
-  is $dom->tree->[5][1], ' HTML4 ',             'right comment';
-  is $dom->tree->[7][1], ' bad idea -- HTML4 ', 'right comment';
+  is $dom->tree->[1][1], ' HTML5 ',                      'right comment';
+  is $dom->tree->[3][1], ' bad idea -- HTML5 ',          'right comment';
+  is $dom->tree->[5][1], '<',                            'wrong comment';
+  is $dom->tree->[6][1], "!-- HTML4 -- >\n",             'wrong comment';
+  is $dom->tree->[7][1], '<',                            'wrong comment';
+  is $dom->tree->[8][1], "!-- bad idea -- HTML4 -- >\n", 'wrong comment';
+
+  for ('<!-->', '<!--->', '<!-- --!>') {
```

Comment:
`$_` with such a wide scope is not good style.

Reply:
In fact, i don't like loops in tests at all, they only make errors harder to find because line numbers get so much less useful.